pkg/server/plugin/bundlepublisher/exec/exec_test.go

```diff
@@ -0,0 +1,185 @@
+package exec
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"testing"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire-plugin-sdk/pluginsdk/support/bundleformat"
+	bundlepublisherv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/server/bundlepublisher/v1"
+	"github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/types"
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	"github.com/spiffe/spire/pkg/common/catalog"
+	"github.com/spiffe/spire/test/plugintest"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/spiffe/spire/test/util"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+)
+
+func TestConfigure(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+
+		configureRequest *configv1.ConfigureRequest
+		newClientErr     error
+		expectCode       codes.Code
+		expectMsg        string
+		config           *Config
+	}{
+		{
+			name: "success",
+			config: &Config{
+				Cmd:    "cat",
+				Args:   []string{},
+				Format: "spiffe",
+			},
+		},
+		{
+			name: "no cmd",
+			config: &Config{
+				Format: "spiffe",
+			},
+			expectCode: codes.InvalidArgument,
+			expectMsg:  "unable to decode configuration: At -",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			options := []plugintest.Option{
+				plugintest.CaptureConfigureError(&err),
+				plugintest.CoreConfig(catalog.CoreConfig{
+					TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
+				}),
+				plugintest.ConfigureJSON(tt.config),
+			}
+
+			p := newPlugin()
+
+			plugintest.Load(t, builtin(p), nil, options...)
+			spiretest.RequireGRPCStatusHasPrefix(t, err, tt.expectCode, tt.expectMsg)
+
+			if tt.expectMsg != "" {
+				require.Nil(t, p.config)
+				return
+			}
+
+			// Check that the plugin has the expected configuration.
+			tt.config.bundleFormat, err = bundleformat.FromString(tt.config.Format)
+			require.NoError(t, err)
+			require.Equal(t, tt.config, p.config)
+		})
+	}
+}
+
+func TestPublishBundle(t *testing.T) {
+	testBundle := getTestBundle(t)
+
+	for _, tt := range []struct {
+		name string
+
+		newClientErr error
+		expectCode   codes.Code
+		expectMsg    string
+		config       *Config
+		bundle       *types.Bundle
+		putObjectErr error
+	}{
+		{
+			name:   "success",
+			bundle: testBundle,
+			config: &Config{
```

Comment:
This test is not verifying if we are really sending bundles to some place, what do you think about saving the response in a temporal file? or create a fake for `exec.CommandContext` using hook and store whatever command is send to it?
with this we can be sure we are executing what is expected